File: src/main/java/com/davidm1a2/afraidofthedark/common/spell/component/property/FloatSpellComponentPropertyBuilder.java
package com.davidm1a2.afraidofthedark.common.spell.component.property;

import org.apache.commons.lang3.Validate;

/**
 * Builder for float spell component property
 */
public class FloatSpellComponentPropertyBuilder extends BoundedSpellComponentPropertyBuilder<Float, FloatSpellComponentPropertyBuilder>
{
    /**
     * Builds the spell component property
     *
     * @return The built spell component property
     */
    public SpellComponentProperty build()
    {
        Validate.notNull(name);
        Validate.notNull(description);
        Validate.notNull(setter);
        Validate.notNull(getter);
        Validate.notNull(defaultValue);

        return new FloatSpellComponentProperty(name, description, setter, getter, defaultValue, minValue, maxValue);
    }
}

File: src/main/kotlin/com/davidm1a2/afraidofthedark/common/spell/component/property/FloatSpellComponentPropertyBuilder.kt
package com.davidm1a2.afraidofthedark.common.spell.component.property

/**
 * Builder for float spell component property
 */
class FloatSpellComponentPropertyBuilder :
    BoundedSpellComponentPropertyBuilder<Float, FloatSpellComponentPropertyBuilder>() {
    /**
     * Builds the spell component property
     *
     * @return The built spell component property
     */
    override fun build(): SpellComponentProperty<Float> {
        return FloatSpellComponentProperty(
            baseName!!,
            setter!!,
            getter!!,
            defaultValue!!,
            minValue,
            maxValue
        )
    }
}